independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/catalog/json/JsonPlatform.java

```diff
@@ -61,6 +62,23 @@ public void addStream(PlatformStream stream) {
         streams.put(stream.getId(), stream);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        JsonPlatform that = (JsonPlatform) o;
+        return Objects.equals(platformKey, that.platformKey);
```

Comment:
TBH, i'm not sure about this equals impl. You could just compare the keys directly instead. What is it meant to hepl with?

Reply:
That is necessary when the JsonPlatform is added to a Collection (Set).